classes/services/NavigationMenuService.inc.php

```diff
@@ -0,0 +1,433 @@
+<?php
+
+/**
+ * @file classes/services/NavigationMenuService.php
+ *
+ * Copyright (c) 2014-2017 Simon Fraser University
+ * Copyright (c) 2000-2017 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class NavigationMenuService
+ * @ingroup services
+ *
+ * @brief Helper class that encapsulates NavigationMenu business logic
+ */
+
+namespace PKP\Services;
+
+class NavigationMenuService {
+
+	public function getMenuItemTypes() {
+
+		$types = array(
+			NMI_TYPE_ABOUT => __('navigation.about'),
+			NMI_TYPE_SUBMISSIONS => __('navigation.submissions'),
+			NMI_TYPE_CURRENT => __('site.journalCurrent'),
+			NMI_TYPE_ABOUT_CONTEXT => __('about.aboutContext'),
+			NMI_TYPE_EDITORIAL_TEAM => __('about.editorialTeam'),
+			NMI_TYPE_CONTACT => __('about.contact'),
+			NMI_TYPE_ANNOUNCEMENTS => __('manager.announcements'),
+			NMI_TYPE_CUSTOM => __('manager.navigationMenus.customType'),
+
+			NMI_TYPE_USER_LOGOUT => __('user.logOut'),
+			NMI_TYPE_USER_LOGOUT_AS => __('user.logOutAs'),
+			NMI_TYPE_USER_PROFILE => __('common.viewProfile'),
+			NMI_TYPE_ADMINISTRATION => __('navigation.admin'),
+			NMI_TYPE_USER_DASHBOARD => __('navigation.dashboard'),
+			NMI_TYPE_USER_REGISTER => __('navigation.register'),
+			NMI_TYPE_USER_LOGIN => __('navigation.login'),
+
+		);
+
+		\HookRegistry::call('NavigationMenus::itemTypes', array(&$types));
+
+		return $types;
+	}
+
+	function getDisplayStatus(&$navigationMenuItem) {
+		$request = \Application::getRequest();
+		$dispatcher = $request->getDispatcher();
+
+		$isUserLoggedIn = \Validation::isLoggedIn();
+		$isUserLoggedInAs = \Validation::isLoggedInAs();
+		$context = $request->getContext();
+
+		// Set navigationMenuItem type display template
+		$templateMgr = \TemplateManager::getManager(\Application::getRequest());
+		$templateReplaceTitle = $templateMgr->get_template_vars($navigationMenuItem->getLocalizedTitle());
+		if ($templateReplaceTitle) {
+			$navigationMenuItem->setTitle();
+			$navigationMenuItem->setTitle($templateReplaceTitle, \AppLocale::getLocale());
+		}
```

Comment:
If we can, I'd like to further limit this so that the user must specify `{$loggedInUserName}` with the brakcets and `$`. That will prevent unwanted clashes. 

If we can do a bit of regex to extract the `{$string}` and replace it, it would be more flexible and be used with `Log out as {$loggedInUsername}`.

Reply:
OK (for one {$parameter})